api/src/main/java/org/eclipse/microprofile/jwt/JWTCallerPrincipal.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.eclipse.microprofile.jwt;
+
+import org.keycloak.representations.JsonWebToken;
+
+import javax.security.enterprise.CallerPrincipal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A standard CallerPrincipal implementation that provides access to the JWT claims that are required by
+ * the microprofile token.
+ */
+public class JWTCallerPrincipal extends CallerPrincipal {
+    /** TODO: need to decide on a JWT parsing/verification library or create one */
+    private JsonWebToken jwt;
```

Comment:
Each impl needs to decide that, not the spec.  The spec may require a base interface to be satisfied, but please avoid using Keycloak in the spec.